nodes/image_processing_utils.py

```diff
@@ -0,0 +1,158 @@
+
+    good_buckets = np.array([[0], [np.pi / 2]])
+    min_include = -1
+
+    half_bucket = np.pi * 5 / 180
+
+    # First split the angles into buckets 90 degrees apart. Find the set of buckets that include the most points.
+    # This is a course move through the state space buckets are 10 degrees wide spaced 5 degrees apart.
+
+    for theta_0 in np.linspace(0, np.pi / 2.0, 18, endpoint=False):
+        buckets = [[], [], []]
+        for angle in angles:
+
+            if theta_0 - half_bucket <= angle[0] <= theta_0 + half_bucket:
+                buckets[0].append(angle)
+            elif theta_0 - half_bucket - np.pi / 2.0 <= angle[0] <= theta_0 + half_bucket - np.pi / 2.0:
+                buckets[1].append(angle)
+            elif theta_0 - half_bucket + np.pi / 2.0 <= angle[0] <= theta_0 + half_bucket + np.pi / 2.0:
+                buckets[2].append(angle)
+
+        include = buckets[0].__len__() + buckets[1].__len__()
+
+        if include > min_include:
+            min_include = include
+            good_buckets = copy.copy(buckets)
+
+    #  Now we take a mean value to find the actual angle from the course buckets.
+
+    theta = 0
+
+    n = 0.0
+
+    if good_buckets[0].__len__() != 0:
+        theta += np.average(good_buckets[0])
+        n += 1.0
+    if good_buckets[1].__len__() != 0:
+        theta += np.average(good_buckets[1]) - np.pi / 2
+        n += 1.0
+    if good_buckets[2].__len__() != 0:
+        theta += np.average(good_buckets[2]) + np.pi / 2
+        n += 1.0
+
+    if n != 0:
+        theta *= 1 / n
+
+    #  Remake buckets at the refined center with indices.
+    final_bucket = [[], []]
+
+    for i in range(angles.__len__()):
+        if theta - half_bucket <= angles[i][0] <= theta + half_bucket:
+            final_bucket[0].append(i)
+        elif theta - half_bucket - np.pi / 2.0 <= angles[i][0] <= theta + half_bucket - np.pi / 2.0 or \
+                theta - half_bucket + np.pi / 2.0 <= angles[i][0] <= theta + half_bucket + np.pi / 2.0:
+            final_bucket[1].append(i)
+
+    headings = np.array([[theta], [theta + np.pi / 2.0]])
+    clusters = np.array(final_bucket)
+
+    # Normalize angles (pi=0, pi/2=-pi/2)
+    headings = np.array(list([[heading[0], heading[0] - np.pi][heading[0] > np.pi / 2]] for heading in headings))
+
+    if abs(headings[0]) > abs(headings[1]):
+        headings = headings[::-1]
+        clusters = clusters[::-1]
+
+    #  There is an ambiguity in which direction is "forward" when theta is close to pi/2.
+    #  -1 is right turn, 1 is left turn.
+    turn_bias = -1  # Starts off at -1 could be changed by advanced algorithms
+    # but probably shouldn't for this naive approach.
+
+    # The margin of what is considered "close" to an angle
+    tolerance = np.pi / 60  # 3 degrees
+
+    head_abs = np.abs(headings[1])
+    if np.abs(head_abs - 1.57) < tolerance:
+        headings[1] = turn_bias * head_abs
+
+    return headings, clusters
+
+
+def find_crosswalk(img):
```

Comment:
Add some comment: less about functionality and more about what it DOES once it finds the crosswalk (or how it returns it)